### PR TITLE
0.9.0 (2) - wrap hook functionality from ApolloClient instance

### PR DESCRIPTION
**Pull request by phryneas**
This PR builds on https://github.com/apollographql/apollo-client/pull/11617 and does the same as #215 or #206, but should tree-shake better than #215 and has better UX than #206

### CODE REVIEW

File: packages/client-react-streaming/src/DataTransportAbstraction/index.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ export {
   useSuspenseQuery,
   useReadQuery,
   useBackgroundQuery,
-} from "./hooks.js";
+} from "@apollo/client/index.js";
```
Comment (phryneas) [inline packages/client-react-streaming/src/DataTransportAbstraction/index.ts:10]: No more custom implementations & it doesn't matter where it's imported from 🎉

Comment (jerelmiller) [reply packages/client-react-streaming/src/DataTransportAbstraction/index.ts:10]: I'm curious, is there any reason not to just deprecate these exports and have users import directly from `@apollo/client`? That could simplify things even further 🙂

File: packages/client-react-streaming/src/DataTransportAbstraction/WrappedApolloClient.tsx [hunk 2]
```diff
@@ -26,13 +27,24 @@ type SimulatedQueryInfo = {
...
     super(options);
+
+    if (!(this.cache instanceof WrappedInMemoryCache)) {
+      throw new Error(
+        "When using Apollo Client streaming SSR, you must use the `InMemoryCache` variant provided by the streaming package."
+      );
+    }
   }
 
+  [wrappers] = hookWrappers;
```
Comment (jerelmiller) [inline packages/client-react-streaming/src/DataTransportAbstraction/WrappedApolloClient.tsx:42]: See https://github.com/apollographql/apollo-client/pull/11617/files#r1500875764

I'd be curious if we could store this on `QueryManager` which would make a path for `useReadQuery` to work in all scenarios.

**Review by jerelmiller — APPROVED** [review]
I like this a lot! Really creative solution to allow for more simplification.

File: packages/client-react-streaming/src/DataTransportAbstraction/index.ts (hunk 1 above)
Comment (phryneas) [reply packages/client-react-streaming/src/DataTransportAbstraction/index.ts:10]: Removing it from this package, but keeping it in the "outer" package for now. (Will drop it in a future version.)

**Review by jerelmiller — APPROVED** [review]

**Review by jerelmiller — APPROVED** [review]

**Pull request merged**